### PR TITLE
[IMP]Improved code to make the availibity of the snippet as much as avai...

### PR DESCRIPTION
**Pull request by psa-odoo**
...lable, once you drag and drop a snippet, no way to remove it..

### CODE REVIEW

File: addons/website_event/controllers/main.py [hunk 1]
```diff
@@ -272,3 +272,16 @@ def registration_confirm(self, event, **post):
             'attendees': attendees,
             'event': event,
         })
+
+    @http.route([
+        "/event/registration_badge_designer/<int:event_id>"
+    ], type='http', auth="public", website=True)
```
Comment (vme-odoo) [inline addons/website_event/controllers/main.py]: Why auth public here ??

File: addons/event/event.py [hunk 2]
```diff
@@ -364,3 +367,31 @@ def message_get_suggested_recipients(self):
             if attendee.partner_id:
                 self._message_add_suggested_recipient(recipients, attendee, partner=attendee.partner_id, reason=_('Customer'))
         return recipients
+
+    @api.multi
+    def action_send_event_badge(self):
+        """ Open a window to compose an email, with the template - 'event_badge'
+            message loaded by default
+        """
+        assert len(self) == 1, 'This option should only be used for a single id at a time.'
```
Comment (jem-odoo) [inline addons/event/event.py]: use `self.ensure_one()` instead

File: addons/website/static/src/css/website.css [hunk 3]
```diff
@@ -974,3 +974,7 @@ div.carousel .item.text_only .container > .carousel-image {
 #oe_main_menu_navbar .navbar-nav.navbar-right:last-child {
   margin-right: 0 !important;
 }
+
+.container nav#oe_main_menu_navbar {
```
Comment (jem-odoo) [inline addons/website/static/src/css/website.css]: why this change ?

File: addons/website_event/models/event.py [hunk 4]
```diff
@@ -93,3 +93,11 @@ def _track_subtype(self, init_values):
         elif 'website_published' in init_values and not self.website_published:
             return 'website_event.mt_event_unpublished'
         return super(event, self)._track_subtype(init_values)
+
+    @api.multi
+    def open_event_badge_designer(self):
```
Comment (jem-odoo) [inline addons/website_event/models/event.py]: call it `action_open_badge_edition`

File: addons/website_event/models/event.py [hunk 5]
```diff
@@ -93,3 +93,11 @@ def _track_subtype(self, init_values):
...
+        return {
```
Comment (jem-odoo) [inline addons/website_event/models/event.py]: add a `self.ensure_one()` too

File: addons/website_event/views/website_event_design_registration_badge.xml [hunk 6]
```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+    <data>
+        <template id="report_design_registration_badge">
```
Comment (jem-odoo) [inline addons/website_event/views/website_event_design_registration_badge.xml]: Should be in `report/` directory, in a file named `report/event_event_templates.xml`.

File: addons/website_event/controllers/main.py [hunk 7]
```diff
@@ -275,3 +275,16 @@ def registration_confirm(self, event, **post):
...
         })
+
+    @http.route([
+        "/event/registration_badge_designer/<int:event_id>"
+    ], type='http', auth="user", website=True)
+    def view_designer(self, event_id, **post):
+        dummy, action = request.registry.get('ir.model.data').get_object_reference(
+            request.cr, request.uid, 'event', 'action_event_view')
+        values = {
+            'event_obj': request.registry.get('event.event').browse(
```
Comment (jem-odoo) [inline addons/website_event/controllers/main.py]: simply call it `event` (change it in the template too) to avoid confusion with the old convention for `event_obj = self.pool.get('event.event')`

File: addons/event/email_template.xml [hunk 8]
```diff
@@ -2,6 +2,22 @@
 <openerp>
     <data>
 
+        <record id="event_badge" model="mail.template">
```
Comment (jem-odoo) [inline addons/event/email_template.xml]: id = `event_registration_badge`

File: addons/website_event/controllers/main.py [hunk 9]
```diff
@@ -275,3 +275,16 @@ def registration_confirm(self, event, **post):
...
+                request.cr, request.uid, event_id),
+            'action': action
```
Comment (jem-odoo) [inline addons/website_event/controllers/main.py]: if `action` contains only the id of the action, you should call it `action_id`

File: addons/website_event/views/website_event_report.xml [hunk 10]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+    <data>
```
Comment (jem-odoo) [inline addons/website_event/views/website_event_report.xml]: remove `data` tag, and use ` <odoo>` instead of `<openerp>`

File: addons/event_sale/models/event.py [hunk 11]
```diff
@@ -13,7 +13,6 @@ class event_event(models.Model):
         'event.event.ticket', 'event_id', string='Event Ticket',
         default=lambda rec: rec._default_tickets(), copy=True)
 
-    badge_back = fields.Html('Badge Back', translate=True, states={'done': [('readonly', True)]})
     badge_innerleft = fields.Html('Badge Innner Left', translate=True, states={'done': [('readonly', True)]})
     badge_innerright = fields.Html('Badge Inner Right', translate=True, states={'done': [('readonly', True)]})
```
Comment (jem-odoo) [inline addons/event_sale/models/event.py]: Shouldn't all the badge fields grouped in the same module ? Or is there a special reason for these 2 fields to be in `event_sale` ?
@tde-banana-odoo your opinion ?

File: addons/website/static/src/css/website.css (hunk 3 above)
Comment (psa-odoo) [inline addons/website/static/src/css/website.css]: The screencast display the reason for the change.

screenshot from 2015 04 29 19 02 45

File: addons/event_sale/models/event.py (hunk 11 above)
Comment (psa-odoo) [inline addons/event_sale/models/event.py]: Reason for moving the field 'badge_back' into event is to make the right side badge of report to make editable when only event installed.

**Comment by psa-odoo** [comment]
@jem-odoo  
All the changes of your comments are done in https://github.com/odoo/enterprise/pull/181  so plz have a look their.
